lib/common/schemas.c

```diff
@@ -279,14 +280,38 @@ wrap_libxslt(bool finalize)
  * \param[in] entry  Directory entry whose filename to check
  *
  * \return 1 if the entry's filename is of the form
- *         <tt>upgrade-X.Y-ORDER.xsl</tt>, or 0 otherwise
+ *         <tt>upgrade-X.Y-ORDER.xsl</tt> with each number in the range 0 to
+ *         255, or 0 otherwise
  */
 static int
 transform_filter(const struct dirent *entry)
 {
-    const char *re = "upgrade-[[:digit:]]+\\.[[:digit:]]+-[[:digit:]]+\\.xsl";
+    const char *re = NULL;
+    unsigned int major = 0;
+    unsigned int minor = 0;
+    unsigned int order = 0;
 
-    return pcmk__str_eq(entry->d_name, re, pcmk__str_regex)? 1 : 0;
+    /* Each number is an unsigned char, which is 1 to 3 digits long. (Pacemaker
```

Comment:
ignoring possibility of zero-padded numbers because that's a stupid concern TBH; we're in control of these filenames